### PR TITLE
11 get reviews queries

### PR DESCRIPTION
**Pull request by teyahbd**
Completed ticket 11! Unsure if additional error test needed for undefined query.

### CODE REVIEW

File: models/reviews.models.js [hunk 1]
```diff
@@ -45,30 +46,66 @@ exports.fetchCommentsByReviewId = (review_id) => {
...
 };
 
-exports.fetchReviews = (category) => {
+exports.fetchReviews = ({ category, sort_by, order }) => {
   const validCategoryValues = [
     "euro_game",
     "social_deduction",
     "dexterity",
     "childrens_games",
+    undefined,
```
Comment (two-triangles) [inline models/reviews.models.js]: This wouldn't be a valid value, there is a better way we could handle this x 3(in relation to your other green lists)

File: models/reviews.models.js [hunk 2]
```diff
@@ -45,30 +46,66 @@ exports.fetchCommentsByReviewId = (review_id) => {
...
+    category = category.replace(/_/g, " ");
     queryStr += ` WHERE reviews.category=$1`;
     queryValues.push(category);
   }
 
-  queryStr += ` GROUP BY reviews.review_id
-  ORDER BY reviews.created_at DESC;`;
+  queryStr += ` GROUP BY reviews.review_id`;
+
+  if (sort_by !== undefined) {
```
Comment (two-triangles) [inline models/reviews.models.js]: this isn't very dry, we can make the query string more dynamic. If we are handling invalid sort_by queries and order queries before we get to this point, we can be safe in the knowledge that if we don't error out, they must be valid values and we can insert them into our query string :)

File: models/reviews.models.js [hunk 3]
```diff
@@ -45,30 +46,66 @@ exports.fetchCommentsByReviewId = (review_id) => {
...
+    "category",
+    "created_at",
+    "votes",
+    undefined,
+  ];
+
+  const validOrderValues = ["asc", "desc", undefined];
+
+  if (
+    !validCategoryValues.includes(category) ||
```
Comment (two-triangles) [inline models/reviews.models.js]: a none existent category would be a 404 - as it could exist in the future?

File: models/reviews.models.js [hunk 4]
```diff
@@ -45,30 +46,66 @@ exports.fetchCommentsByReviewId = (review_id) => {
     });
 };
 
-exports.fetchReviews = (category) => {
+exports.fetchReviews = ({ category, sort_by, order }) => {
```
Comment (two-triangles) [inline models/reviews.models.js]: We could give the paramters default values to make your conditional logic dryer

File: models/categories.models.js [hunk 5]
```diff
@@ -5,3 +5,24 @@ exports.fetchCategories = () => {
     return categories;
   });
 };
+
+exports.fetchCategoryBySlug = (slug) => {
```
Comment (teyahbd) [inline models/categories.models.js:9]: okay so I added the promise all here to stop the case where a category exists but has no reviews from returning an error. So checking that the returned query is an empty array but also that the category doesn't exist in the list of all categories. Maybe there's a better way to do it than this?

File: controllers/reviews.controllers.js [hunk 6]
```diff
@@ -38,9 +39,16 @@ exports.getCommentsByReviewId = (req, res, next) => {
...
 
 exports.getReviews = (req, res, next) => {
-  fetchReviews(req.query.category)
-    .then((reviews) => {
-      res.status(200).send(reviews);
+  const { category, sort_by, order } = req.query;
+  const promises = [fetchReviews(category, sort_by, order)];
+
+  if (category) {
+    promises.push(fetchCategoryBySlug(category));
```
Comment (corkesw) [inline controllers/reviews.controllers.js:46]: As fetchCategoryBySlug is a utility function we should really be unit testing this. You would have to query the database in your tests to confirm that this works as expected.

**Review by corkesw — COMMENTED** [review]
Nice work!

**Pull request merged**